entitymanager.js

```diff
@@ -0,0 +1,44 @@
+function EntityManager() {
+  var entities = {};
+  var nextId = 0;
+
+  this.add = function(entity) {
+    entity.registerId(nextId);
+    entities[nextId++] = entity;
+  }
+
+  this.update = function() {
+    for (var index in entities) {
+      if (entities[index].update()) {
+        // Deletes the index property from the entities object.
+        delete entities[index];
+      }
+    }
+  }
+
+  this.render = function() {
+    for (var index in entities) {
+      entities[index].render();
+    }
+  }
+
+  this.checkCollisions = function() {
+    var x = 1;
+    for (var i in entities) {
```

Comment:
This is not possible as entities is not an array but an object. I do this so every entity has a consistent id in the entitymanager, aka even if an entity dies the objects that were made after it don't change their ids.